itou/approvals/models.py

```diff
@@ -230,8 +230,10 @@ def with_assigned_company(self):
         )
 
     def is_assigned_to(self, company_id):
+        from itou.job_applications.models import JobApplication
+
         return (
-            self.filter(user__job_applications__to_company=company_id)
+            self.filter(Exists(JobApplication.objects.filter(approval=OuterRef("pk"), to_company=company_id)))
```

Comment:
De ce que je comprend ça va faire la sous-requête pour chaque ligne donc dans le cas présent ça serait pas plus efficace d'utiliser d'un `IN` ?
```suggestion
            self.filter(pk__in=Subquery(JobApplication.objects.filter(to_company=company_id).values("approval")))
```

Reply:
Et bien... c’est équivalent :joy: 
Le EXPLAIN de DjDT donne le même plan d’exécution, peu importe la forme. J’ai inclus la requête dans les deux cas, où on voit bien soit le `IN` soit le `EXISTS()`. Je suppose que PostgreSQL est malin et prend l’opération la plus efficace pour réaliser la requête, peu importe la forme demandée.

[exists.sql.gz](https://github.com/gip-inclusion/les-emplois/files/15248624/exists.sql.gz)
[in.sql.gz](https://github.com/gip-inclusion/les-emplois/files/15248625/in.sql.gz)

J’ai préféré ta version qui me semble effectivement plus simple :+1:

Reply:
Yep mais à mon avis ça dépend aussi de ce que tu lui demandes à coté, là c'est probablement lié à l'annotation `assigned_company` qui tape sur la même table et donc il optimise le travail, sur un truc plus simple c'est pas dit.